src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArray = [];
+  let newState = { ...state };
```

Comment:
The variable name `newState` is a bit misleading as it suggests a new state is created each time, but it's actually being mutated with each action. A better name might be `currentState` or `stateCopy` to reflect that it's a mutable copy of the initial state.